### ISSUE TITLE
[core]Using multiple models can cause `new_x`  lag feature shift

### ISSUE THREAD

**Issue by MrTangsai** [Nixtla/mlforecast#289]
### What happened + What you expected to happen

Hi, 
1. `models` param in `MLForecast` class could receive multi models as a list , then use `predict` method, may cause lag feature drift, therefore, different models may get diferent features
2. I would expect that different models use the same lag features at the same prediction moment. If I just use date_features, it works well, there confused me when I use lag features or lag_transform features
3. the same as above, When  I call `predict` method second time, it's lag features still drift on the basis of the first time, but I just use one model, everything is OK.

### Versions / Dependencies

1. `mlforecast`: 0.11.3
2. `python`: 3.8.16
3. `os`: Ubuntu

### Reproduction script

python code is below:
```python
import pandas as pd
from mlforecast import MLForecast
from mlforecast.target_transforms import Differences
from sklearn.linear_model import LinearRegression
from xgboost import XGBRegressor
from lightgbm import LGBMRegressor
from IPython.display import display
def inspect_input(new_x):
    """Displays the model inputs to inspect them"""
    display(new_x)
    return new_x
df = pd.read_csv('https://datasets-nixtla.s3.amazonaws.com/air-passengers.csv', parse_dates=['ds'])
fcst = MLForecast(
    models=[LinearRegression(), XGBRegressor(), LGBMRegressor(verbose=-1)],
    freq='MS',  # our serie has a monthly frequency
    lags=[12],
    target_transforms=[Differences([1])],
)
fcst.fit(df, )
fcst.predict(1, before_predict_callback=inspect_input)
```
then the result is below:
<img width="762" alt="image" src="https://github.com/Nixtla/mlforecast/assets/27365618/2f18a759-d3a3-470c-a740-b3f92011416f">

### Issue Severity

High: It blocks me from completing my task.

**Comment by jmoralez** [comment]
Hey @MrTangsai, thanks for using mlforecast and for the excellent report. I believe this was introduced in #255 and released in 0.11.0, so if this is blocking you you can install a prior version and it should work as expected. We'll work on a fix soon.

**Comment by MrTangsai** [comment]
thanks a lot! it works well in new version.
